pkg/frontend/querymiddleware/model.proto

```diff
@@ -87,6 +87,9 @@ message Extent  {
   reserved 3;
   string trace_id = 4 [(gogoproto.jsontag) = "-"];
   google.protobuf.Any response = 5 [(gogoproto.jsontag) = "response"];
+  // Unix timestamp (millis) when the query with this response was executed.
+  // If the response is combination of multiple queries over time, queryTime is timestamp of oldest query.
+  int64 queryTime = 6;
```

Comment:
[nit] In other protos (e.g. `mimir.proto`), we call it "timestamp_ms". You may want to consider `query_timestamp_ms` for consistency.